integration-test/148-sea-ocean-labels-water-layer.py

```diff
@@ -10,33 +10,163 @@
                                u'name:de': u'Gr\xf6nlandsee', u'source': u'openstreetmap.org', u'name:fr': u'Mer du Groenland', u'name:zh': u'\u683c\u9675\u5170\u6d77', u'name:sk': u'Gr\xf3nske more', u'name:lt': u'Grenlandijos j\u016bra', u'gns:uni': u'-2149780', u'name:uk': u'\u0413\u0440\u0435\u043d\u043b\u0430\u043d\u0434\u0441\u044c\u043a\u0435 \u043c\u043e\u0440\u0435', u'name:sv': u'Gr\xf6nlandshavet', u'name:hu': u'Gr\xf6nlandi-tenger', u'name:hr': u'Grenlandsko more', u'name:eo': u'Gronlanda Maro', u'name:en': u'Greenland Sea', u'name': u'Greenland Sea', u'place': u'sea'}))
+
+        # Seas are not graded by area, they are constant zoom (that should probably change)
         self.assert_has_feature(
             z, x, y, 'water',
-            {'kind': 'sea', 'name': 'Greenland Sea',
+            {'kind': 'sea', 'name': 'Greenland Sea', 'min_zoom': 5.0,
              'label_placement': True})
+
+        # even though OSM tags as place, Tilezen recasts to water layer
         self.assert_no_matching_feature(
             z, x, y, 'places',
             {'kind': 'sea', 'name': 'Greenland Sea'})
 
-# NOTE: No ocean points in the North America extract :-(
+        # ensure the zoom range is correct vis-a-vis the size
+        x, y = deg2num(1.0, 1.0, 4)
+        self.assert_no_matching_feature(
+            4, x, y, 'water',
+            {'kind': 'sea', 'name': 'Greenland Sea'})
+        x, y = deg2num(1.0, 1.0, 5)
+        self.assert_has_feature(
+            5, x, y, 'water',
+            {'kind': 'sea', 'name': 'Greenland Sea', 'min_zoom': 5.0,
+             'label_placement': True})
+
+    def test_hudson_bay_large_polygon(self):
+        z = 16
+        lon, lat = (1, 1)
+
+        x, y = deg2num(lat, lon, z)
+        # without remapping min_zoom this area would result in min_zoom of 6,
+        # but after remapping it'd be 2
+        area = 80000000001
+
+        self.generate_fixtures(
+            dsl.way(9441240, dsl.box_area(z, x, y, area), {
+                'name': u'Hudson Bay',
+                'place': u'sea',
+                'wikidata': u'Q3040',
+                'source': u'openstreetmap.org',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'water',
+            {'kind': 'sea', 'name': 'Hudson Bay', 'min_zoom': 2.0,
+             'label_placement': True})
+
+        # ensure the zoom range is correct vis-a-vis the size
+        x, y = deg2num(1.0, 1.0, 1)
+        self.assert_no_matching_feature(
+            1, x, y, 'water',
+            {'kind': 'sea', 'name': 'Hudson Bay',
+             'label_placement': True})
+        # Even though the feature has min_zoom 2, it isn't exported in a tile
```

Comment:
This was super confusing so added a note – basically even though the feature _should_ be in a zoom 2 tile because it's min_zoom is 2, for build performance and tile size reasons we actually exclude it elsewhere until zoom 4 tiles.